### PR TITLE
Musum/limit number of config models build when adding node

### PR DESCRIPTION
**Pull request by hmusum**
First 3 commits are just refactorings, last one is the actual change

### CODE REVIEW

File: configserver/src/main/java/com/yahoo/vespa/config/server/modelfactory/ModelsBuilder.java [hunk 1]
```diff
@@ -204,7 +202,7 @@ private Version findLatest(Set<Version> versionSet) {
     private boolean mayBeUsedOn(AllocatedHosts hosts, Version version) {
```
Comment (bratseth) [inline configserver/src/main/java/com/yahoo/vespa/config/server/modelfactory/ModelsBuilder.java]: Should rename to isUsedOn now ...

**Review by bratseth — COMMENTED** [review]
Are we sure we never end up with an empty set of versions now?

File: configserver/src/main/java/com/yahoo/vespa/config/server/modelfactory/ModelsBuilder.java (hunk 1 above)
Comment (hmusum) [reply configserver/src/main/java/com/yahoo/vespa/config/server/modelfactory/ModelsBuilder.java]: Done

**Comment by bratseth** [comment]
Ok, great

**Pull request merged**